### PR TITLE
AOSP libs support

### LINKED ISSUE #64: Add zxing to Android.bp

### LINKED ISSUE #67: Add Android.bp for custom CameraX AARs

### PR DESCRIPTION
**Pull request by SebaUbuntu**
Closes #64
Closes #67

### CODE REVIEW

**Comment by luk1337** [comment]
Would prefer to leave gradle as is.

**Comment by SebaUbuntu** [comment]
> Would prefer to leave gradle as is.

I just keep them in sync to be sure we use the same libraries for both Android Studio and AOSP, but I can just remove Gradle handling

**Comment by SebaUbuntu** [comment]
> Would prefer to leave gradle as is.

Made it optional and disabled Gradle updating

File: app/update_libs.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+        new_build_gradle = []
+        in_deps_list = False
+        for line in build_gradle:
+            if in_deps_list:
+                if line == "}":
+                    in_deps_list = False
+                    new_build_gradle.append(line)
+                continue
+            else:
```
Comment (luk1337) [inline app/update_libs.py:303]: elif

File: app/update_libs.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    new_android_bp = []
+    in_deps_list = False
+    for line in android_bp:
+        if in_deps_list:
+            if line == "    ],":
+                in_deps_list = False
+                new_android_bp.append(line)
+            continue
+        else:
+            if line == "    static_libs: [":
```
Comment (luk1337) [inline app/update_libs.py]: elif

File: app/libs.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    ["androidx.camera", "camera-view", CAMERAX_VERSION, "luk_camerax"],
+
+    # Coil
+    ["io.coil-kt", "coil", "2.2.0", "maven_central"],
+    ["io.coil-kt", "coil-base", "2.2.0", "maven_central"],
+    ["io.coil-kt", "coil-video", "2.2.0", "maven_central"],
+
+    # Coil dependencies
+    ["androidx.lifecycle", "lifecycle-common-java8", "2.5.1", "sdk"],
+    ["com.google.errorprone", "error_prone_annotations", "2.15.0", "sdk"],
```
Comment (luca020400) [inline app/libs.py]: Gradle doesn't pull that

File: app/libs.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    ["com.squareup.okhttp3", "okhttp", "4.9.0", "maven_central"],
```
Comment (luca020400) [inline app/libs.py]: Gradle pulls in 4.10.0

File: app/libs.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+# [project, module, version, repo]
+# sdk: not a local library, will use AOSP prebuilts/sdk one
+# Note than for non-SDK libs you must manually specify its dependencies
+DEPS: List[List[str]] = [
+    # Core
+    ["androidx.core", "core-ktx", "1.8.0", "sdk"],
+    ["androidx.appcompat", "appcompat", "1.5.0", "sdk"],
+
+    # Base
+    ["androidx.concurrent", "concurrent-futures-ktx", "1.1.0", "gmaven"],
```
Comment (luca020400) [inline app/libs.py]: Gradle doesn't pull that

File: app/libs.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    ["androidx.constraintlayout", "constraintlayout", "2.1.4", "sdk"],
+    ["androidx.exifinterface", "exifinterface", "1.3.3", "sdk"],
```
Comment (luca020400) [inline app/libs.py]: Gradle doesn't pull that

File: app/libs.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    ["io.coil-kt", "coil-video", "2.2.0", "maven_central"],
+
+    # Coil dependencies
+    ["androidx.lifecycle", "lifecycle-common-java8", "2.5.1", "sdk"],
+    ["com.google.errorprone", "error_prone_annotations", "2.15.0", "sdk"],
+    ["com.squareup.okhttp3", "okhttp", "4.9.0", "maven_central"],
+    ["com.squareup.okio", "okio", "3.2.0", "maven_central"],
+    ["com.squareup.okio", "okio-jvm", "3.2.0", "maven_central"],
+    ["org.jetbrains", "annotations", "13.0", "sdk"],
+    ["org.jetbrains.kotlin", "kotlin-android-extensions-runtime", "1.7.10", "maven_central"],
```
Comment (luca020400) [inline app/libs.py]: Gradle doesn't pull that

File: app/libs.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    ["org.jetbrains.kotlin", "kotlin-parcelize-runtime", "1.7.10", "maven_central"],
```
Comment (luca020400) [inline app/libs.py]: Gradle doesn't pull that

**Pull request closed without merging**